R/get_latest_resource.R

```diff
@@ -0,0 +1,105 @@
+#' Get the latest resource from a data set
+#'
+#' Returns the latest resource available in a dataset.
+#'
+#' There are some datasets on the open data platform that
+#' keep historic resources instead of updating existing ones.
+#' For these it is useful to be able to retrieve the latest
+#' resource. As of 5.7.2024 these data sets include:
+#' * gp-practice-populations
+#' * gp-practice-contact-details-and-list-sizes
+#' * nhsscotland-payments-to-general-practice
+#' * dental-practices-and-patient-registrations
+#' * general-practitioner-contact-details
+#' * prescribed-dispensed
+#' * prescriptions-in-the-community
+#' * community-pharmacy-contractor-activity
+#'
+#' @param dataset_name name of the dataset as found on
+#' \href{https://www.opendata.nhs.scot/}{NHS Open Data platform}
+#' @param rows (optional) specify the max number of rows to return.
+#' @param row_filters (optional) a named list or vector that specifies values of
+#'  columns/fields to keep.
+#' e.g. list(Date = 20220216, Sex = "Female").
+#' @param col_select (optional) a character vector containing the names of
+#' desired columns/fields.
+#' e.g. c("Date", "Sex").
+#' @param include_context (optional) If `TRUE` additional information about the
+#' resource will be added as columns to the data, including the resource ID, the
+#' resource name, the creation date and the last modified/updated date.
```

Comment:
This simplifies the maintenance of the documentation and ensures it's consistent.

```suggestion
@inheritParams get_resource
```

You can actually make functions share a documentation page, which might be appropriate in this case - https://r-pkgs.org/man.html#sec-man-multiple-functions

Reply:
Thanks. I did think about this but get_latest_resource needsa dataset name instead of a res_id. I wasn't sure if there was a way to inherit all parameters except for one.

Reply:
It will only inherit names that match, so this should be fine.